flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontendParser.java

```diff
@@ -451,4 +479,25 @@ public static InfoOptions parseInfoCommand(String[] args) throws CliArgsExceptio
 		}
 	}
 
+	public static MainOptions parseMainCommand(String[] args) throws CliArgsException {
+
+		// drop all arguments after an action
+		final List<String> params= Arrays.asList(args);
+		for (String action: CliFrontend.ACTIONS) {
+			int index = params.indexOf(action);
+			if(index != -1) {
+				args = Arrays.copyOfRange(args, 0, index);
+				break;
+			}
+		}
+
+		try {
+			DefaultParser parser = new DefaultParser();
+			CommandLine line = parser.parse(MAIN_OPTIONS, args, false);
```

Comment:
Actually you can use `CommandLine line = parser.parse(MAIN_OPTIONS, args, true);` to halt once you reach a non-arg.